src/PixelFinder.h

```diff
@@ -0,0 +1,176 @@
+/*
+ * LSST Data Management System
+ * Copyright 2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SPHGEOM_PIXELFINDER_H_
+#define LSST_SPHGEOM_PIXELFINDER_H_
+
+/// \file
+/// \brief This file provides a base class for pixel finders.
+
+#include "lsst/sphgeom/RangeSet.h"
+
+#include "ConvexPolygonImpl.h"
+
+
+namespace lsst {
+namespace sphgeom {
+namespace {
```

Comment:
Mainly because anonymous namespace in a header file looks strange :smile: I'm not entirely sure what the purpose of that could be. Anonymous namespace becomes unique in every .cc file that includes this header, which also make classes contained in that namespace unique.  The `PixelFinder` class in file A.cc will be totally unrelated to the `PixelFinder` class in B.cc (even if they have all the same template parameters), and this could be confusing (why do you want to have multiple "copies" of the same class?)